### ISSUE TITLE
Read env from docker secrets 

### ISSUE THREAD

**Issue by archfurry** [postalsys/emailengine#146]
**Is your feature request related to a problem? Please describe.**
There is a secret storage mechanism in Docker Swarm that mounts secret values as a file. It is much safer way to provide a secret to docker container then env variable.
Many docker project now support _FILE postfix for every supported env variable.
For example EmailEngine supports EENGINE_PREPARED_LICENSE. It is a sensitive information. It would be convenient for Docker Swarm setup to provide env called EENGINE_PREPARED_LICENSE_FILE: '/run/secrets/ee_license' 

**Describe the solution you'd like**
For my projects I use following TS snippet:
```
import { readFileSync } from 'fs';
import randomString from 'randomstring';
function readEnv(
  envName: string,
  defaultValue: string | (() => string)
): string {
  try {
    const fileVal = process.env[envName + '_FILE'];
    if (fileVal !== undefined) {
      const buf = readFileSync(fileVal);
      return buf.toString();
    }
    const envVal = process.env[envName];
    if (envVal !== undefined) {
      return envVal;
    }
    throw 'Using defaultValue';
  } catch (e) {
    if (typeof defaultValue === 'string') {
      return defaultValue;
    }
    return defaultValue();
  }
}
```

Notable use cases are:
1) env with default value
`const port = readEnv('PORT', '3000');`
2) env with randomly generated secret value
`const cookieSecret = readEnv('COOKIE_SECRET', () => randomString.generate(100) );`
3) required env with no default value
`const encSecret = readEnv('ENC_SECRET', () => { throw new Error('ENC_SECRET is not set!') } );`

**Describe alternatives you've considered**
There is a way to provide .env file to working directory of the EmailEngine but it complicates handling of secrets elsewhere and adds extra challenges of generating such files

**Additional context**
I have no additional context. Thank you for Your attention!

**Comment by andris9** [comment]
Sounds reasonable. Is there any kind of official or semi-official documentation about this convention?

**Comment by archfurry** [comment]
The first package I saw that uses _FILE counterpart on any env variable was LEGO [Docs](https://go-acme.github.io/lego/):

> The environment variable names can be suffixed by _FILE to reference a file instead of a value. More information [here](https://go-acme.github.io/lego/dns/#configuration-and-credentials).

This docker image uses a lot of secret values so it becomes necessary to store them using Docker Swarm secrets. 

Also [this](https://docs.docker.com/engine/swarm/secrets/#build-support-for-docker-secrets-into-your-images) Docker manual recommends to add support of _FILE envs. It mentions some images like mysql and wordpress which support this feature without breaking backward compatibility.
But [wordpress](https://hub.docker.com/_/wordpress) and [mysql](https://hub.docker.com/_/mysql) uses only limited set of sensitive envs:

> As an alternative to passing sensitive information via environment variables, _FILE may be appended to the previously listed environment variables, causing the initialization script to load the values for those variables from files present in the container. In particular, this can be used to load passwords from Docker secrets stored in /run/secrets/<secret_name> files.
...
Currently, this is supported for WORDPRESS_DB_HOST, WORDPRESS_DB_USER, WORDPRESS_DB_PASSWORD...

So in case of EmailEngine there sould be also a limited set?
I could not find any reliable information on this subject other than mentioned above.

**Comment by andris9** [comment]
Thanks. I think I'll add the `_FILE` support for every supported env value.

**Comment by andris9** [comment]
Added by https://github.com/postalsys/emailengine/pull/147 and will be included in the next release

**Comment by andris9** [comment]
Released as part of v2.22.4, [docs](https://emailengine.app/configuration#file-env)
